InSituLib/VtkMappedMeshSource.cpp

```diff
@@ -78,29 +78,75 @@ int VtkMappedMeshSource::RequestData(vtkInformation *,
 	if (outInfo->Get(vtkStreamingDemandDrivenPipeline::UPDATE_PIECE_NUMBER()) > 0)
 		return 1;
 
+	// Points
 	this->Points->Reset();
-
 	vtkNew<VtkMeshNodalCoordinatesTemplate<double> > nodeCoords;
 	nodeCoords->SetNodes(_mesh->getNodes());
 	this->Points->SetData(nodeCoords.GetPointer());
 
-	// TODO nodal vals
-
+	// Elements
 	vtkNew<VtkMappedMesh> elems;
 	elems->GetImplementation()->SetNodes(_mesh->getNodes());
 	elems->GetImplementation()->SetElements(_mesh->getElements());
-
-	 // Use the mapped point container for the block points
 	elems->SetPoints(this->Points.GetPointer());
-
 	output->Allocate(elems->GetNumberOfCells());
 	output->ShallowCopy(elems.GetPointer());
 
-	// Mapped data array for material ids
-	vtkNew<VtkMappedElementDataArrayTemplate<unsigned> > materialIds;
-	materialIds->SetElements(&_mesh->getElements(), _mesh->getNElements());
-	materialIds->SetName("MaterialIDs");
-	output->GetCellData()->AddArray(materialIds.GetPointer());
+	// Arrays
+	MeshLib::Properties const & properties = _mesh->getProperties();
+	std::vector<std::string> const& propertyNames = properties.getPropertyNames();
+
+	for(std::vector<std::string>::const_iterator it = propertyNames.cbegin(); it != propertyNames.cend(); ++it)
+	{
+		if (addDoubleProperty(properties, *it))
+			continue;
+
+		if (addIntProperty(properties, *it))
+			continue;
+
+		DBUG ("Mesh property \"%s\" with unknown data type.", *it->c_str());
+	}
+
+	output->GetPointData()->ShallowCopy(this->PointData.GetPointer());
+	output->GetCellData()->ShallowCopy(this->CellData.GetPointer());
+
+	return 1;
+}
+
+int VtkMappedMeshSource::addDoubleProperty(MeshLib::Properties const& properties,
+                                           std::string const& prop_name) const
+{
+	boost::optional<MeshLib::PropertyVector<double> const &> propertyVector(properties.getProperty<double>(prop_name));
+	if(!propertyVector)
+		return 0;
+
+	vtkNew<VtkMappedPropertyVectorTemplate<double> > dataArray;
+	dataArray->SetPropertyVector(const_cast<MeshLib::PropertyVector<double> &>(*propertyVector));
+	dataArray->SetName(prop_name.c_str());
+
+	if(propertyVector->getMeshItemType() == MeshLib::MeshItemType::Node)
+		this->PointData->AddArray(dataArray.GetPointer());
+	else if(propertyVector->getMeshItemType() == MeshLib::MeshItemType::Cell)
+		this->CellData->AddArray(dataArray.GetPointer());
+
+	return 1;
+}
+
+int VtkMappedMeshSource::addIntProperty(MeshLib::Properties const& properties,
```

Comment:
It looks for me that there is no difference in addDoubleProperty() and addIntProperty() except for data type. maybe you can define a template function and call it like `addProperty<double>()`, `addProperty<int>()`?